core/payment/src/models/allocation.rs

```diff
@@ -15,6 +15,7 @@ pub struct WriteObj {
     pub total_amount: BigDecimalField,
     pub spent_amount: BigDecimalField,
     pub remaining_amount: BigDecimalField,
+    pub timestamp: NaiveDateTime,
```

Comment:
Nah, we don't put `timestamp`s in `WriteObj`s. The database is supposed to assign these.

Reply:
Later in the file `ReadObj = WriteObj`

also database was not friendly in generating the dates when i tried

Reply:
> Later in the file ReadObj = WriteObj

You should define a separate type `ReadObj` without the timestamp. That's how it's done for other models.

> also database was not friendly in generating the dates when i tried

Maybe the switch to `DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW'))` suggested above wold help. DB-generated timestamps seem to work all fine for events.

Reply:
> You should define a separate type ReadObj without the timestamp. That's how it's done for other models.
Done!

> Maybe the switch to DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')) suggested above wold help
Indeed, its working fine now :)